examples/window-app/telink/src/AppTask.cpp

```diff
@@ -0,0 +1,631 @@
+
+    // Initialize CHIP server
+#if CONFIG_CHIP_FACTORY_DATA
+    ReturnErrorOnFailure(mFactoryDataProvider.Init());
+    SetDeviceInstanceInfoProvider(&mFactoryDataProvider);
+    SetDeviceAttestationCredentialsProvider(&mFactoryDataProvider);
+    SetCommissionableDataProvider(&mFactoryDataProvider);
+    // Read EnableKey from the factory data.
+    MutableByteSpan enableKey(sTestEventTriggerEnableKey);
+    err = mFactoryDataProvider.GetEnableKey(enableKey);
+    if (err != CHIP_NO_ERROR)
+    {
+        LOG_ERR("GetEnableKey fail");
+        memset(sTestEventTriggerEnableKey, 0, sizeof(sTestEventTriggerEnableKey));
+    }
+#else
+    SetDeviceAttestationCredentialsProvider(Examples::GetExampleDACProvider());
+#endif
+
+    static CommonCaseDeviceServerInitParams initParams;
+    (void) initParams.InitializeStaticResourcesBeforeServerInit();
+    ReturnErrorOnFailure(chip::Server::GetInstance().Init(initParams));
+
+    gExampleDeviceInfoProvider.SetStorageDelegate(&Server::GetInstance().GetPersistentStorage());
+    chip::DeviceLayer::SetDeviceInfoProvider(&gExampleDeviceInfoProvider);
+
+#if CONFIG_CHIP_OTA_REQUESTOR
+    InitBasicOTARequestor();
+#endif
+
+    ConfigurationMgr().LogDeviceConfig();
+    PrintOnboardingCodes(chip::RendezvousInformationFlags(chip::RendezvousInformationFlag::kBLE));
+
+    // Add CHIP event handler and start CHIP thread.
+    // Note that all the initialization code should happen prior to this point to avoid data races
+    // between the main and the CHIP threads.
+    PlatformMgr().AddEventHandler(ChipEventHandler, 0);
+
+    err = ConnectivityMgr().SetBLEDeviceName("TelinkWindow");
+    if (err != CHIP_NO_ERROR)
+    {
+        LOG_ERR("SetBLEDeviceName fail");
+        return err;
+    }
+
+    err = chip::Server::GetInstance().GetFabricTable().AddFabricDelegate(new AppFabricTableDelegate);
+    if (err != CHIP_NO_ERROR)
+    {
+        LOG_ERR("AppFabricTableDelegate fail");
+        return err;
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR AppTask::StartApp(void)
+{
+    CHIP_ERROR err = Init();
+
+    if (err != CHIP_NO_ERROR)
+    {
+        LOG_ERR("AppTask Init fail");
+        return err;
+    }
+
+    AppEvent event = {};
+
+    while (true)
+    {
+        k_msgq_get(&sAppEventQueue, &event, K_FOREVER);
+        DispatchEvent(&event);
+    }
+}
+
+void AppTask::OpenActionAndToggleMoveTypeButtonEventHandler(void)
+{
+    AppEvent event;
+
+    event.Type               = AppEvent::kEventType_Button;
+    event.ButtonEvent.Action = kButtonPushEvent;
+    event.Handler            = ToggleMoveTypeHandler;
+    sAppTask.PostEvent(&event);
+}
+
+void AppTask::CloseActionButtonEventHandler(void)
+{
+    AppEvent event;
+
+    event.Type               = AppEvent::kEventType_Button;
+    event.ButtonEvent.Action = kButtonPushEvent;
+    event.Handler            = CloseHandler;
+    sAppTask.PostEvent(&event);
+}
+
+void AppTask::IdentifyEffectHandler(EmberAfIdentifyEffectIdentifier aEffect)

```

Comment:
Is there nothing we can do to handle the identify in common fashion ?

Reply:
Our Identify functionality is a bit more complicated and implemented per spec to be able to pass the tests (`[TC-I-2.2] Primary functionality with server as DUT`, `[TC-I-2.2] Primary functionality with server as DUT`)